### DISCUSSION TITLE
How do I call a method with a generic return type?

### DISCUSSION THREAD

**Discussion by philpax** [sebastienros/jint#1147] [Q&A]
Hello again!

I've looked through #127, #814, and the tests for #1103 and #1106, but I haven't found an answer to this (unless I'm missing something obvious).

I have a method `public T? ReadColumn<T>(int column)`; how do I call this method and specify what `T` should be? Most of the existing generic support seems to be catered towards T being a type generic or otherwise deducible from the arguments; I'm not sure how to call it when that's not the case.

Appreciate the help, as always 🙏

**Comment by lahma** [comment]
Unfortunately I think that's quite impossible. If the JS call site is valid JS like `const result = obj.ReadColumn(1);` it's quite impossible to know what the generic parameter type should be.

You probably have to options from the top of my head.

1. Create wrapper for each needed method and add to prototype for example, like `const result = obj.ReadStringColumn(1);` which then would go to CLR method that uses `ReadColumn<string>(index)`.
2. Create generic wrapper with needed type  information, `obj.ReadColumn("System.String, System", 1)` or some other alias that you map to type then.

Of course if there's an non-generic version like `public object? ReadColumn(int column)` Jint will try to wrap the result the best it can.
